### ISSUE TITLE
[Suggestion]: Sync wind direction with WilderWild

### ISSUE THREAD

**Issue by vico93** [RandomMcSomethin/fallingleaves#65]
Recently, [WilderWild](https://github.com/FrozenBlock/WilderWild) ([MR](https://modrinth.com/mod/wilder-wild)) added a interesting wind feature that will orient both wind particles and tumbleweeds direction, and since FallingLeaves also have an wind option i thought if would be possible to FL to orientate its leaves particles accordingly to WW wind.

According to Lunade (that implemented this feature on WW) on FrozenBlock's Discord, he added a mixin for that on WW side, so theorically would be possible to FL pull the information to use on its end.

**Comment by AViewFromTheTop** [comment]
Wind itself is implemented through FrozenLib, and is just utilized frequently in Wilder Wild.

**Comment by Fourmisain** [comment]
I tried syncing the wind direction but am not sure if it really makes things better.

Wilder Wild wind seems a lot more subtle than Falling Leaves wind.
I compared campfire smoke with some leaves and the smoke is barely affected while the leaves are flying by, which isn't exactly what one would expect.
Here's a rough illustration of what I mean:
![illustration](https://github.com/RandomMcSomethin/fallingleaves/assets/8464472/b8d88224-f8d1-4db2-bda3-a6b239451342)

Needs more testing/comparisons with other particles, I uploaded a [test version](https://github.com/RandomMcSomethin/fallingleaves/releases/tag/wilderwild-wind-sync-test) if you wanna give it a shot.

**Comment by vico93** [comment]
WilderWild's wind can be tweaked in config i think. Maybe adding a multiplier to the value read from WW/FrozenLib to make it more stronger in FallingLeaves side?

**Comment by Fourmisain** [comment]
I mean the values can be tweaked in code just fine, it's not a technical issue, it's a conceptual one.
The leaves were always more on the fast moving side _by design_. Slowing them down doesn't look too great.
The smoke just isn't moving that fast, _by design_, so you'll always have this discrepancy.
(And changing the behavior of Wilder Wild is out of question.)

It might also not be an actual issue. I only looked at campfires for a minute and just thought it feels a bit odd.
Arguably the screenshot above might not even be that "unrealistic", since leaves can curl and glide even when there's no wind.

So best thing is to just play with it and see how it feels.
(And I'm currently not really playing Minecraft, which is why e.g. #49 is still on the waiting list.)

**Comment by AViewFromTheTop** [comment]
I personally didn’t want to make smoke and whatnot move too fast as there’s not really a simple way to make it spread out, on top of being a large change from Vanilla if I had made it more intense.
Leaves moving faster than most other particles makes sense in my opinion.
Also, would anyone be able to test this with Wind Charges and the Breeze? It should work out of the box but I just want to see if it realistically works.

**Comment by Fourmisain** [comment]
> I personally didn’t want to make smoke and whatnot move too fast as there’s not really a simple way to make it spread out, on top of being a large change from Vanilla if I had made it more intense.

Yeah, that makes sense, I argued similarly against (fast) moving smoke and fireworks in #20.
(Also, I didn't even realize we had a Wilder Wild dev here! 😄)

> Also, would anyone be able to test this with Wind Charges and the Breeze? It should work out of the box but I just want to see if it realistically works.

Oh, do Wind Charges change FrozenLib wind? If so, that's super cool!
I just tried it a bunch with some leaves and campfires but it didn't seem to have any effect, not any explosive effect that I could notice at least.

P.S.
The changes in test version 2 were kinda wrong, the wind coefficient was already accounted for in the original wind speed, so I just reverted that change. Maybe the variance on those per-particle coefficients is too low, need to do some controlled tests.

**Comment by AViewFromTheTop** [comment]
I forgot to mention, I’m glad wind was moved to FrozenLib long ago because I’m aiming to move wind stuff into a separate mod that will be embedded into Wilder Wild, and still use FrozenLib’s system.
Treetrain1 made a Wind Extension class which is how WW controls its specific stuff, and also impacts the main controller in FrozenLib.
So regardless, this should work just fine after WW and Wind split into two mods as long as you’re strictly referencing FrozenLib for this.
The split will very likely be some time after 1.21 releases, I haven’t started work on it yet.

**Comment by AViewFromTheTop** [comment]
Also, FrozenLib’s config has an option to turn off Wind client-side while on a server that doesn’t have FrozenLib installed. I’d keep that in mind as well.
Do you have these changes on a branch by chance? I’m up to help out with this if possible.

**Comment by Fourmisain** [comment]
I've put it on the [wilderwild branch](https://github.com/RandomMcSomethin/fallingleaves/tree/wilderwild), the code's very minimal.
Note that the thing currently doesn't build without a physical FrozenLib jar.
Wanted to depend on it via jitpack, but the build apparently broke over time. For one the jitpack.yml fails due to [using an outdated install script](https://jitpack.io/com/github/FrozenBlock/FrozenLib/1.7.2-mc1.20.6/build.log), which is a [simple fix](https://github.com/Fourmisain/FrozenLib/commit/f580ebc72f25034a1da4be0fc50e8ef6729ddf65)*, but then it fails due to what looks like [some mapping issue with Cloth Config](https://jitpack.io/com/github/Fourmisain/FrozenLib/f580ebc72f/build.log) or something...?
The exact same happens when I try building it locally (and on [GitHub actions as well](https://github.com/Fourmisain/FrozenLib/actions/runs/9242368279/job/25425096853)).
Tried to [fix the build](https://github.com/Fourmisain/FrozenLib/commit/1c6c4c8ce7e27af2f655d49460973aaae7a573a9) by downgrading to loom 1.6 ('+' defaults to 1.7-alpha) and updating the mappings but it didn't seem to help.

Wait, maybe it did?
The [GH action succeeded](https://github.com/Fourmisain/FrozenLib/actions/runs/9246832483), so does a local build after running `gradle clean build --refresh-dependencies`, but [the jitpack build still fails](https://jitpack.io/com/github/Fourmisain/FrozenLib/1c6c4c8ce7/build.log) because... it doesn't find the build artifact? Hm... not sure how to fix that one.

\* Actually, it looks like jitpacks defaults to running `sdk install java 21.0.2-open;sdk use java 21.0.2-open` now? Maybe the jitpack.yml could actually be deleted, though I guess it's still better to hardcore the Java version.

**Comment by Fourmisain** [comment]
k, fixed and PR'd.

Except jitpack likes to run into [read timeouts](https://github.com/RandomMcSomethin/fallingleaves/actions/runs/9250924571) right now, which is weird because locally it works just fine.
Gonna rerun the build in a few hours.

Aaand I just now realize I could probably have used the modrinth maven thingamajig...

**Comment by AViewFromTheTop** [comment]
I was just about to bring up Modrinth lol

**Comment by AViewFromTheTop** [comment]
i'm gonna see if i can just do the ww and wind split soon
